rosidl_generator_c/cmake/rosidl_generator_c_generate_interfaces.cmake

```diff
@@ -116,81 +118,81 @@ if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
     C_STANDARD 11
     COMPILE_OPTIONS -Wall -Wextra -Wpedantic)
 endif()
-if(WIN32)
-  target_compile_definitions(${rosidl_generate_interfaces_TARGET}${_target_suffix}
-    PRIVATE "ROSIDL_GENERATOR_C_BUILDING_DLL_${PROJECT_NAME}")
-endif()
+target_compile_definitions(${rosidl_generate_interfaces_TARGET}${_target_suffix}
+  PRIVATE "ROSIDL_GENERATOR_C_BUILDING_DLL_${PROJECT_NAME}")
```

Comment:
@sloretz nit: `ROSIDL_GENERATOR_C_BUILDING_DLL_*` macros are only used in visibility headers on Windows (e.g. see [here](https://github.com/ros2/rosidl/blob/67bcca666d818823aba8ea08c12160a9badf4e41/rosidl_generator_c/resource/rosidl_generator_c__visibility_control.h.in#L23)).

Reply:
Oh oops. I read that header wrong.

Interestingly CMake has a target property for this already called [`DEFINE_SYMBOL`](https://cmake.org/cmake/help/v3.10/prop_tgt/DEFINE_SYMBOL.html). It adds the compiler definition `<target_name>_EXPORTS` by default, setting the property changes it. I changed it to use that property instead in 519066b